library/core/src/option.rs

```diff
@@ -963,9 +963,11 @@ impl<T> Option<T> {
     /// assert_eq!(y, None);
     /// ```
     #[inline]
+    #[rustc_const_unstable(feature = "const_option", issue = "67441")]
     #[stable(feature = "rust1", since = "1.0.0")]
-    pub fn take(&mut self) -> Option<T> {
-        mem::take(self)
+    pub const fn take(&mut self) -> Option<T> {
+        // TODO replace `mem::replace` by `mem::take` when the latter is const ready
```

Comment:
the TODO [is a reminder for an owner of a draft something to do *before* the PR lands](https://rustc-dev-guide.rust-lang.org/conventions.html#use-todo-comments-for-things-you-dont-want-to-forget), so you want `FIXME` instead.